File: Sources/App/Models/PlaylistEntry.swift
//
//  PlaylistEntry.swift
//  FinchServer
//
//  Created by Stefan Klein Nulent on 22/02/2025.
//

import Vapor
import Fluent

final class PlaylistEntry: Model, @unchecked Sendable {
    
    // MARK: - Constants
    
    static let schema = "playlist_entries"
    
    
    
    // MARK: - Properties
    
    @ID(custom: .id, generatedBy: .database)
    var id: Int?
    
    @Field(key: "index")
    var index: Int
    
    @Parent(key: "playlist_id")
    var playlist: Playlist
    
    @Field(key: "item_id")
    var itemId: Int
    
    @Timestamp(key: "created_at", on: .create)
    var createdAt: Date?
    
    
    
    // MARK: - Construction
    
    init() {}
    
    init(item: Item, index: Int) throws {
        itemId = try item.requireID()
        self.index = index
    }
}

File: FinchServer/Database/PlaylistEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using FinchServer.Beets;

namespace FinchServer.Database;

[Table("playlist_entries")]
public class PlaylistEntry {
    
    // - Properties
    
    [Key]
    public int Id { get; set; }
    
    public required int Index { get; set; }
    
    public required int ItemId { get; set; }

    public DateTime CreatedAt { get; set; }
    
    public Playlist Playlist { get; set; } = null!;
}